File: app/src/androidTest/java/de/hdc/kspchecklist/data/DataIOTest.java
package de.hdc.kspchecklist.data;

import android.content.Context;
import android.support.test.InstrumentationRegistry;

import org.junit.After;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

import java.util.ArrayList;

/**
 * Created by DerTroglodyt on 2017-02-13 12:52.
 * Email example@gmail.com
 * Copyright by HDC, Germany
 */
public class DataIOTest {
    @Before
    public void setUp() throws Exception {

    }

    @After
    public void tearDown() throws Exception {

    }

    @Test
    public void readAssetFile() throws Exception {
        // Context of the app under test.
        Context appContext = InstrumentationRegistry.getTargetContext();

        ArrayList<CheckListItem> list = DataIO.readAssetFile(appContext, "eins.txt");
    }

    @Test
    public void writeLocalFile() throws Exception {
        Context appContext = InstrumentationRegistry.getTargetContext();

        ArrayList<CheckListItem> list = new ArrayList<>();
        list.add(CheckListItem.create("eins", true));
        list.add(CheckListItem.create("zwei", false));
        list.add(CheckListItem.create("drei", true));
        DataIO.writeLocalFile(appContext, "eins.txt", list);

        ArrayList<CheckListItem> list2 = DataIO.readLocalFile(appContext, "eins.txt");
        Assert.assertTrue(list.size() == 3);
    }
}

File: app/src/androidTest/java/de/hdc/kspchecklist/data/DataIOTest.kt
package de.hdc.kspchecklist.data

import android.support.test.InstrumentationRegistry
import org.junit.After
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import java.util.*

/**
 * Created by DerTroglodyt on 2017-02-13 12:52.
 * Email example@gmail.com
 * Copyright by HDC, Germany
 */
class DataIOTest {
    @Before
    @Throws(Exception::class)
    fun setUp() {

    }

    @After
    @Throws(Exception::class)
    fun tearDown() {

    }

    @Test
    @Throws(Exception::class)
    fun readAssetFile() {
        // Context of the app under test.
        val appContext = InstrumentationRegistry.getTargetContext()

        DataIO.readAssetFile(appContext.assets, "Simple.txt")
    }

    @Test
    @Throws(Exception::class)
    fun writeLocalFile() {
        val appContext = InstrumentationRegistry.getTargetContext()

        val list = ArrayList<CheckListItem>()
        list.add(CheckListItem("eins", true))
        list.add(CheckListItem("zwei", false))
        list.add(CheckListItem("drei", true))
        DataIO.writeLocalFile(appContext, "eins.txt", list)

        val list2 = DataIO.readLocalFile(appContext, "eins.txt")
        Assert.assertTrue(list2.size == 3)
    }
}
